build.js

```diff
@@ -1714,6 +1742,16 @@ var getSomeDependencyTriggeredRebuild = function(subpackageNames, resultsCache)
   return someDependencyTriggeredRebuild;
 };
 
+var getLinkingRelatedArgs = function(foreignDependencies, packageResource, buildConfig) {
+  if (foreignDependencies.length === 0) {
+    return [];
+  }
+
+  return (buildConfig.compiler === 'byte' ? ['-custom'] : []).concat(foreignDependencies.map(function(sourceFile) {
```

Comment:
Yeah... It's because of the way the byte code links handles foreign dependencies, you need to build a custom runtime when making the library. This is explained here: http://caml.inria.fr/pub/docs/manual-ocaml-4.00/manual033.html#toc142 at section `19.1.3`. The tl;dr is that the compiler has a table of C functions that the ocaml code is using, so when compiling with foreign C functions (more than the standard lib) it requires you to build a custom runtime with a custom table right inside the library file. For some reason you need to explicitly pass in the flag `-custom` to tell the compiler to do that.